src/index/txindex.cpp

```diff
@@ -13,6 +13,8 @@
 
 constexpr int64_t SYNC_LOG_INTERVAL = 30; // seconds
 
+std::unique_ptr<TxIndex> g_txindex;
```

Comment:
It'd be nice to keep these kinds of globals in the same place. At least g_connman is in init.cpp and exposed as extern in net.h, so you could duplicate that (which is "correct" in that init "owns" the object).

Reply:
That's what I did originally, but that caused problems when building tests. The test binary is not linked against init.cpp, it's linked against test_bitcoin_main.cpp, which is why `g_connman` is also defined there. That took me an annoying amount of time to figure out, and by the time I did I was so pissed off I decided to move it to `txindex.cpp`.

This whole thing where stuff in init has to get redefined in `test_bitcoin_main.cpp` if it's used in the tests seems pretty gross. Do you really thing that's the right way to do it?

Reply:
Hmm, yea, thats a bit gross. Still, if thats what we have now, maybe add a comment noting that in both places and at least be consistent?